File: backend/FinAI.Backend/Entities/Receipt.cs
using System.ComponentModel.DataAnnotations;

namespace FinAI.Backend.Entities;

public class Receipt
{
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    public Guid UserId { get; set; }

    public string ImageUrl { get; set; } = string.Empty;

    public string MerchantName { get; set; } = string.Empty;

    public decimal TotalAmount { get; set; }

    public string Category { get; set; } = "Diğer";

    public string ExtractedDataJson { get; set; } = "{}";

    public Guid? TransactionId { get; set; }

    public string Status { get; set; } = "PENDING"; // PENDING, SAVED, REJECTED

    public double ConfidenceScore { get; set; } = 0.9;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public User User { get; set; } = null!;
}

File: frontend/types/receipt.ts
export interface ParsedReceiptItem {
  name: string;
  price: number;
}

export interface ReceiptResult {
  receiptId: string;
  merchantName: string;
  totalAmount: number;
  category: string;
  dateStr: string;
  itemsJson: string;
  confidenceScore: number;
}

export interface ReceiptHistoryItem {
  id: string;
  merchantName: string;
  totalAmount: number;
  category: string;
  status: string;
  confidenceScore: number;
  createdAt: string;
}
